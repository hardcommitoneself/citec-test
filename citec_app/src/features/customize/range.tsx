// Copyright (C) 2021-Present CITEC Inc. <https://citecsolutions.com/>
// All rights reserved
//
// This file is part of CITEC Inc. source code.
// This software framework contains the confidential and proprietary information
// of CITEC Inc., its affiliates, and its licensors. Your use of these
// materials is governed by the terms of the Agreement between your organisation
// and CITEC Inc., and any unauthorised use is forbidden. Except as otherwise
// stated in the Agreement, this software framework is for your internal use
// only and may only be shared outside your organisation with the prior written
// permission of CITEC Inc.
// CITEC Inc. source code can not be copied and/or distributed without the express
// permission of CITEC Inc.
import Slider from '@mui/material/Slider'
import { setFactors } from 'store/slices/universe'
import { useDispatch } from 'react-redux'
import { useState } from 'react'
import { Icon } from 'assets/icons'
import { customizePopups } from 'features/popups/customize-popups'
import {
   completeTitle,
   titleHighValue,
   titleLowValue,
} from 'features/utils/complete-titles-customize'

interface RangeProps {
   name: string
   state: number[]
   index?: number
   min?: number
   max?: number
}
export const Range = ({ name, state, index, min, max }: RangeProps) => {
   const dispatch = useDispatch()

   const nameUpperCase = name.charAt(0).toUpperCase() + name.slice(1)

   const [rangeValues, setRangeValues] = useState<number[]>(state)

   const setRangeValueOnRedux = (value: number | number[]) => {
      const payload = {
         index,
         value,
      }
      dispatch(setFactors(payload))
   }

   return (
      <div className="mb-2">
         <div className="flex mb-4 text-[#0BACA9]">
            <h4 className="text-xl flex items-center">
               {nameUpperCase} {completeTitle(name)}
            </h4>

            <button className="ml-2" onClick={() => customizePopups(name)}>
               <Icon.Info />
            </button>
         </div>
         <div className="w-full flex justify-between mt-1 text-[#79828D] text-sm">
            <span>{titleLowValue(name)}</span>
            <span>{titleHighValue(name)}</span>
         </div>
         <div className="px-2">
            <Slider
               value={rangeValues}
               onChangeCommitted={(_, newValue) => setRangeValueOnRedux(newValue)}
               onChange={(_, newValue) => {
                  setRangeValues(newValue as never)
               }}
               valueLabelDisplay="auto"
               min={min ?? 0}
               max={max ?? 100}
            />
         </div>
      </div>
   )
}
