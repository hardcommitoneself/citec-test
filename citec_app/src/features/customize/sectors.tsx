// Copyright (C) 2021-Present CITEC Inc. <https://citecsolutions.com/>
// All rights reserved
//
// This file is part of CITEC Inc. source code.
// This software framework contains the confidential and proprietary information
// of CITEC Inc., its affiliates, and its licensors. Your use of these
// materials is governed by the terms of the Agreement between your organisation
// and CITEC Inc., and any unauthorised use is forbidden. Except as otherwise
// stated in the Agreement, this software framework is for your internal use
// only and may only be shared outside your organisation with the prior written
// permission of CITEC Inc.
// CITEC Inc. source code can not be copied and/or distributed without the express
// permission of CITEC Inc.
import { Checkbox } from '@mui/material'
import { useDispatch } from 'react-redux'
import { useAppSelector } from 'store/hooks'
import { setSectors } from 'store/slices/universe'

export const SectorsUniverse = () => {
   const dispatch = useDispatch()
   const defaultSectors = useAppSelector(
      (state) => state.static_data.universe.sectors
   )
   const sectorsFilter = useAppSelector((state) => state.universe.sectors_filter)

   const handleOnChange = (sector: string) => {
      if (sectorsFilter.includes(sector)) {
         const response = sectorsFilter.filter((e) => e !== sector)
         dispatch(setSectors(response))
      } else {
         const response = [...sectorsFilter]
         response.push(sector)
         dispatch(setSectors(response))
      }
   }

   return (
      <div className="grid grid-cols-4 gap-2">
         {defaultSectors.map((sector, key) => (
            <label className="text-neutral-600 cursor-pointer" key={key}>
               <Checkbox
                  id={`custom-checkbox-${key}`}
                  data-testid={`custom-checkbox-${key}`}
                  name={sector}
                  checked={sectorsFilter.includes(sector)}
                  onChange={() => handleOnChange(sector)}
               />
               {sector}
            </label>
         ))}
      </div>
   )
}
