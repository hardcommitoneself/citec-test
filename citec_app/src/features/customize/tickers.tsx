// Copyright (C) 2021-Present CITEC Inc. <https://citecsolutions.com/>
// All rights reserved
//
// This file is part of CITEC Inc. source code.
// This software framework contains the confidential and proprietary information
// of CITEC Inc., its affiliates, and its licensors. Your use of these
// materials is governed by the terms of the Agreement between your organisation
// and CITEC Inc., and any unauthorised use is forbidden. Except as otherwise
// stated in the Agreement, this software framework is for your internal use
// only and may only be shared outside your organisation with the prior written
// permission of CITEC Inc.
// CITEC Inc. source code can not be copied and/or distributed without the express
// permission of CITEC Inc.
import { useDispatch } from 'react-redux'
import { setTickerToInclude, setTickerToExclude } from 'store/slices/universe'
import { useForm, useValidateTicker } from 'features/utils/hooks'
import { useAppSelector } from 'store/hooks'
import { AutocompleteInput } from 'features/utils/autocomplete'

interface TickersProps {
   type: 'Include' | 'Exclude'
}

export const Tickers = ({ type }: TickersProps) => {
   const dispatch = useDispatch()
   let tickers: any[] = []
   const tickers_include = useAppSelector((state) => state.universe.to_include)
   const tickers_exclude = useAppSelector((state) => state.universe.to_exclude)

   if (type == 'Include') {
      tickers = tickers_include
   } else {
      tickers = tickers_exclude
   }

   const {
      values: { ticker },
      handleInputChange,
      handleResetForm,
   } = useForm({
      ticker: '',
   })

   const addRow = (e: React.SyntheticEvent) => {
      e.preventDefault()

      if (type == 'Include') {
         dispatch(setTickerToInclude([...tickers, ticker]))
      } else {
         dispatch(setTickerToExclude([...tickers, ticker]))
      }
      handleResetForm()
   }

   const handleDeleteTicker = (ticker: string) => {
      const tickerFiltered = tickers.filter(
         (tickersOnRedux) => tickersOnRedux !== ticker
      )
      if (type == 'Include') dispatch(setTickerToInclude(tickerFiltered))
      else dispatch(setTickerToExclude(tickerFiltered))
   }

   const isValidTicker = useValidateTicker(ticker)

   return (
      <div className="mb-10">
         <h3 className="font-bold text-neutral-600 mb-4 text-lg">{type} tickers</h3>

         <form
            className="flex items-center justify-center w-full mb-3 relative"
            autoComplete="off"
            onSubmit={addRow}
         >
            <AutocompleteInput
               onChangeValue={(e) => handleInputChange(e, { toUpperCase: true })}
               value={ticker}
               name="ticker"
               error={isValidTicker || !ticker}
               noMargin
            />

            {!isValidTicker && !!ticker && (
               <span
                  className="text-sm absolute -bottom-6 left-0 text-red-500"
                  data-testid="alert-invalid"
               >
                  Ticker {ticker} is invalid
               </span>
            )}

            <button
               type="submit"
               className="ml-2 bg-[#00A393] hover:bg-green-700 btn-table-ticker"
               disabled={!isValidTicker}
               data-testid="submit-ticker"
            >
               +
            </button>
         </form>

         {tickers.map((ticker, i) => (
            <div className="flex w-full items-center mb-3" key={i}>
               <span className="table-item w-full">{ticker}</span>
               <button
                  onClick={() => handleDeleteTicker(ticker)}
                  className="ml-2 bg-[#E24646] hover:bg-red-700 btn-table-ticker"
                  data-testid="delete-ticker"
               >
                  -
               </button>
            </div>
         ))}
      </div>
   )
}
