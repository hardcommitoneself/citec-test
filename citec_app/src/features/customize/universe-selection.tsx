// Copyright (C) 2021-Present CITEC Inc. <https://citecsolutions.com/>
// All rights reserved
//
// This file is part of CITEC Inc. source code.
// This software framework contains the confidential and proprietary information
// of CITEC Inc., its affiliates, and its licensors. Your use of these
// materials is governed by the terms of the Agreement between your organisation
// and CITEC Inc., and any unauthorised use is forbidden. Except as otherwise
// stated in the Agreement, this software framework is for your internal use
// only and may only be shared outside your organisation with the prior written
// permission of CITEC Inc.
// CITEC Inc. source code can not be copied and/or distributed without the express
// permission of CITEC Inc.

import { useAppSelector } from 'store/hooks'
import { useSelectUniverse } from 'features/utils/hooks/use-select-universe'

export const UniverseSelection = () => {
   const { handleSelectUniverse } = useSelectUniverse()

   const universes = useAppSelector((state) => state.static_data.universe.universes)
   const universeSelected = useAppSelector((state) => state.universe.universe_base)

   return (
      <div className="grid grid-cols-4 w-full mb-2 gap-3">
         {universes.map((universe, key) => (
            <button
               key={key}
               className={
                  universeSelected === universe.name
                     ? 'universe-btn-select'
                     : 'universe-btn'
               }
               onClick={() => handleSelectUniverse(universe.name)}
            >
               {universe?.description}
            </button>
         ))}
      </div>
   )
}
