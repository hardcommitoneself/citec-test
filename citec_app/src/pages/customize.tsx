// Copyright (C) 2021-Present CITEC Inc. <https://citecsolutions.com/>
// All rights reserved
//
// This file is part of CITEC Inc. source code.
// This software framework contains the confidential and proprietary information
// of CITEC Inc., its affiliates, and its licensors. Your use of these
// materials is governed by the terms of the Agreement between your organisation
// and CITEC Inc., and any unauthorised use is forbidden. Except as otherwise
// stated in the Agreement, this software framework is for your internal use
// only and may only be shared outside your organisation with the prior written
// permission of CITEC Inc.
// CITEC Inc. source code can not be copied and/or distributed without the express
// permission of CITEC Inc.

import { Accordion, AccordionDetails, AccordionSummary } from '@mui/material'
import { useEffect } from 'react'
import { useDispatch } from 'react-redux'

import { setSectors } from 'store/slices/universe'
import { useAppSelector } from 'store/hooks'
import { useGetCustomUniverse } from 'features/utils/hooks/use-get-custom-universe'
import { Icon } from 'assets/icons'
import {
   Factors,
   SectorsUniverse,
   Tickers,
   TotalsUniverse,
   UniverseSelection,
} from 'features/customize'

export const Customize = () => {
   const customUniverse = useGetCustomUniverse()

   const dispatch = useDispatch()

   const sectors = useAppSelector((state) => state.static_data.universe.sectors)
   const sectorsFilter = useAppSelector((state) => state.universe.sectors_filter)

   const parameters = [
      { name: 'Market Screener', component: <UniverseSelection /> },
      { name: 'Sector Screener', component: <SectorsUniverse /> },
      { name: 'Smart Beta Factor Screener', component: <Factors /> },
      {
         name: 'Include/Exclude',
         component: (
            <div className="grid grid-cols-2 gap-3">
               <Tickers type="Include" /> <Tickers type="Exclude" />
            </div>
         ),
      },
   ]

   const parametersWithInfo = ['Sector Screener', 'Smart Beta Factor Screener']

   useEffect(() => {
      if (sectorsFilter.length === 0) {
         dispatch(setSectors(sectors))
      }
   }, [])

   return (
      <>
         <TotalsUniverse totals={customUniverse} />

         <div className="my-3">
            {parameters.map(({ name, component }, i) => (
               <Accordion key={i}>
                  <AccordionSummary>
                     <p className="flex items-center text-xl my-2 text-[#2d405a]">
                        <span className="font-semibold mr-2 ">{name}</span>
                        {parametersWithInfo.includes(name) && <Icon.Info />}
                     </p>
                  </AccordionSummary>
                  <AccordionDetails>{component}</AccordionDetails>
               </Accordion>
            ))}
         </div>
      </>
   )
}
