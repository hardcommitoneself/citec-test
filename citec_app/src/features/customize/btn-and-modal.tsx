// Copyright (C) 2021-Present CITEC Inc. <https://citecsolutions.com/>
// All rights reserved
//
// This file is part of CITEC Inc. source code.
// This software framework contains the confidential and proprietary information
// of CITEC Inc., its affiliates, and its licensors. Your use of these
// materials is governed by the terms of the Agreement between your organisation
// and CITEC Inc., and any unauthorised use is forbidden. Except as otherwise
// stated in the Agreement, this software framework is for your internal use
// only and may only be shared outside your organisation with the prior written
// permission of CITEC Inc.
// CITEC Inc. source code can not be copied and/or distributed without the express
// permission of CITEC Inc.

import { Button, Modal } from '@mui/material'
import { Icon } from 'assets/icons'
import { PageContainer } from 'features/ui/page-container'
import { Customize } from 'pages'
import { useState } from 'react'

export const ButtonAndModalCustomize = () => {
   const [openModal, setOpenModal] = useState(false)
   const handleOpen = () => setOpenModal(true)
   const handleClose = () => setOpenModal(false)

   return (
      <>
         <Button id="criteria_customize_button" variant="home" onClick={handleOpen}>
            Customize
         </Button>

         <Modal open={openModal} sx={{ overflowY: 'scroll' }}>
            <div className="absolute top-4 inset-x-[7.5%] bg-[#F4F6FA] rounded-xl py-3 w-[85%]">
               <PageContainer
                  titlePage="Universe summary"
                  actionBtn={
                     <button onClick={handleClose}>
                        <Icon.Xicon />
                     </button>
                  }
               >
                  <Customize />
               </PageContainer>
            </div>
         </Modal>
      </>
   )
}
