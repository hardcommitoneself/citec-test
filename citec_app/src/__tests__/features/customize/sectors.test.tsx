// Copyright (C) 2021-Present CITEC Inc. <https://citecsolutions.com/>
// All rights reserved
//
// This file is part of CITEC Inc. source code.
// This software framework contains the confidential and proprietary information
// of CITEC Inc., its affiliates, and its licensors. Your use of these
// materials is governed by the terms of the Agreement between your organisation
// and CITEC Inc., and any unauthorised use is forbidden. Except as otherwise
// stated in the Agreement, this software framework is for your internal use
// only and may only be shared outside your organisation with the prior written
// permission of CITEC Inc.
// CITEC Inc. source code can not be copied and/or distributed without the express
// permission of CITEC Inc.
import '@testing-library/jest-dom'
import { fireEvent, render, screen } from '@testing-library/react'
import { SectorsUniverse } from 'features/customize'
import { Provider } from 'react-redux'
import configureStore from 'redux-mock-store'

const initialState = {
   static_data: {
      universe: {
         sectors: [
            'Industrials',
            'Financials',
            'Health Care',
            'Consumer Staples',
            'Communication Services',
            'Information Technology',
            'Consumer Discretionary',
            'Materials',
            'Cash',
            'Real Estate',
            'Energy',
            'Utilities',
         ],
      },
   },
   universe: {
      sectors_filter: [
         'Industrials',
         'Financials',
         'Health Care',
         'Consumer Staples',
         'Communication Services',
         'Information Technology',
         'Consumer Discretionary',
         'Materials',
         'Cash',
         'Real Estate',
         'Energy',
         'Utilities',
      ],
   },
}

const initialStateWithoutSectorFilter = {
   static_data: {
      universe: {
         sectors: [
            'Industrials',
            'Financials',
            'Health Care',
            'Consumer Staples',
            'Communication Services',
            'Information Technology',
            'Consumer Discretionary',
            'Materials',
            'Cash',
            'Real Estate',
            'Energy',
            'Utilities',
         ],
      },
   },
   universe: {
      sectors_filter: [],
   },
}

const mockStore = configureStore()
const store = mockStore(initialState)
const storeWithoutSectorFilter = mockStore(initialStateWithoutSectorFilter)

test('test Sectors Universe screening', () => {
   const { container } = render(
      <Provider store={store}>
         <SectorsUniverse />
      </Provider>
   )
   const linkElement = screen.getByText(/Energy/i)
   expect(linkElement).toBeInTheDocument()
   expect(container).toMatchSnapshot()
})

test('click Sector full marked', () => {
   const { getByTestId } = render(
      <Provider store={store}>
         <SectorsUniverse />
      </Provider>
   )

   fireEvent.click(getByTestId('custom-checkbox-5'))
})

test('click Sector without markers', () => {
   const { getByTestId } = render(
      <Provider store={storeWithoutSectorFilter}>
         <SectorsUniverse />
      </Provider>
   )

   fireEvent.click(getByTestId('custom-checkbox-5'))
})
