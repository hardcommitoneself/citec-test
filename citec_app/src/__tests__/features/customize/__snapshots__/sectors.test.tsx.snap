// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test Sectors Universe screening 1`] = `
<div>
  <div
    class="grid grid-cols-4 gap-2"
  >
    <label
      class="text-neutral-600 cursor-pointer"
    >
      <span
        class="MuiCheckbox-root MuiCheckbox-colorPrimary MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary PrivateSwitchBase-root Mui-checked css-12wnr2w-MuiButtonBase-root-MuiCheckbox-root"
        data-testid="custom-checkbox-0"
      >
        <input
          checked=""
          class="PrivateSwitchBase-input css-1m9pwf3"
          data-indeterminate="false"
          id="custom-checkbox-0"
          name="Industrials"
          type="checkbox"
        />
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="CheckBoxIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M19 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.11 0 2-.9 2-2V5c0-1.1-.89-2-2-2zm-9 14l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
          />
        </svg>
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </span>
      Industrials
    </label>
    <label
      class="text-neutral-600 cursor-pointer"
    >
      <span
        class="MuiCheckbox-root MuiCheckbox-colorPrimary MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary PrivateSwitchBase-root Mui-checked css-12wnr2w-MuiButtonBase-root-MuiCheckbox-root"
        data-testid="custom-checkbox-1"
      >
        <input
          checked=""
          class="PrivateSwitchBase-input css-1m9pwf3"
          data-indeterminate="false"
          id="custom-checkbox-1"
          name="Financials"
          type="checkbox"
        />
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="CheckBoxIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M19 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.11 0 2-.9 2-2V5c0-1.1-.89-2-2-2zm-9 14l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
          />
        </svg>
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </span>
      Financials
    </label>
    <label
      class="text-neutral-600 cursor-pointer"
    >
      <span
        class="MuiCheckbox-root MuiCheckbox-colorPrimary MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary PrivateSwitchBase-root Mui-checked css-12wnr2w-MuiButtonBase-root-MuiCheckbox-root"
        data-testid="custom-checkbox-2"
      >
        <input
          checked=""
          class="PrivateSwitchBase-input css-1m9pwf3"
          data-indeterminate="false"
          id="custom-checkbox-2"
          name="Health Care"
          type="checkbox"
        />
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="CheckBoxIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M19 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.11 0 2-.9 2-2V5c0-1.1-.89-2-2-2zm-9 14l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
          />
        </svg>
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </span>
      Health Care
    </label>
    <label
      class="text-neutral-600 cursor-pointer"
    >
      <span
        class="MuiCheckbox-root MuiCheckbox-colorPrimary MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary PrivateSwitchBase-root Mui-checked css-12wnr2w-MuiButtonBase-root-MuiCheckbox-root"
        data-testid="custom-checkbox-3"
      >
        <input
          checked=""
          class="PrivateSwitchBase-input css-1m9pwf3"
          data-indeterminate="false"
          id="custom-checkbox-3"
          name="Consumer Staples"
          type="checkbox"
        />
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="CheckBoxIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M19 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.11 0 2-.9 2-2V5c0-1.1-.89-2-2-2zm-9 14l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
          />
        </svg>
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </span>
      Consumer Staples
    </label>
    <label
      class="text-neutral-600 cursor-pointer"
    >
      <span
        class="MuiCheckbox-root MuiCheckbox-colorPrimary MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary PrivateSwitchBase-root Mui-checked css-12wnr2w-MuiButtonBase-root-MuiCheckbox-root"
        data-testid="custom-checkbox-4"
      >
        <input
          checked=""
          class="PrivateSwitchBase-input css-1m9pwf3"
          data-indeterminate="false"
          id="custom-checkbox-4"
          name="Communication Services"
          type="checkbox"
        />
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="CheckBoxIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M19 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.11 0 2-.9 2-2V5c0-1.1-.89-2-2-2zm-9 14l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
          />
        </svg>
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </span>
      Communication Services
    </label>
    <label
      class="text-neutral-600 cursor-pointer"
    >
      <span
        class="MuiCheckbox-root MuiCheckbox-colorPrimary MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary PrivateSwitchBase-root Mui-checked css-12wnr2w-MuiButtonBase-root-MuiCheckbox-root"
        data-testid="custom-checkbox-5"
      >
        <input
          checked=""
          class="PrivateSwitchBase-input css-1m9pwf3"
          data-indeterminate="false"
          id="custom-checkbox-5"
          name="Information Technology"
          type="checkbox"
        />
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="CheckBoxIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M19 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.11 0 2-.9 2-2V5c0-1.1-.89-2-2-2zm-9 14l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
          />
        </svg>
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </span>
      Information Technology
    </label>
    <label
      class="text-neutral-600 cursor-pointer"
    >
      <span
        class="MuiCheckbox-root MuiCheckbox-colorPrimary MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary PrivateSwitchBase-root Mui-checked css-12wnr2w-MuiButtonBase-root-MuiCheckbox-root"
        data-testid="custom-checkbox-6"
      >
        <input
          checked=""
          class="PrivateSwitchBase-input css-1m9pwf3"
          data-indeterminate="false"
          id="custom-checkbox-6"
          name="Consumer Discretionary"
          type="checkbox"
        />
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="CheckBoxIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M19 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.11 0 2-.9 2-2V5c0-1.1-.89-2-2-2zm-9 14l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
          />
        </svg>
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </span>
      Consumer Discretionary
    </label>
    <label
      class="text-neutral-600 cursor-pointer"
    >
      <span
        class="MuiCheckbox-root MuiCheckbox-colorPrimary MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary PrivateSwitchBase-root Mui-checked css-12wnr2w-MuiButtonBase-root-MuiCheckbox-root"
        data-testid="custom-checkbox-7"
      >
        <input
          checked=""
          class="PrivateSwitchBase-input css-1m9pwf3"
          data-indeterminate="false"
          id="custom-checkbox-7"
          name="Materials"
          type="checkbox"
        />
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="CheckBoxIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M19 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.11 0 2-.9 2-2V5c0-1.1-.89-2-2-2zm-9 14l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
          />
        </svg>
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </span>
      Materials
    </label>
    <label
      class="text-neutral-600 cursor-pointer"
    >
      <span
        class="MuiCheckbox-root MuiCheckbox-colorPrimary MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary PrivateSwitchBase-root Mui-checked css-12wnr2w-MuiButtonBase-root-MuiCheckbox-root"
        data-testid="custom-checkbox-8"
      >
        <input
          checked=""
          class="PrivateSwitchBase-input css-1m9pwf3"
          data-indeterminate="false"
          id="custom-checkbox-8"
          name="Cash"
          type="checkbox"
        />
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="CheckBoxIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M19 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.11 0 2-.9 2-2V5c0-1.1-.89-2-2-2zm-9 14l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
          />
        </svg>
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </span>
      Cash
    </label>
    <label
      class="text-neutral-600 cursor-pointer"
    >
      <span
        class="MuiCheckbox-root MuiCheckbox-colorPrimary MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary PrivateSwitchBase-root Mui-checked css-12wnr2w-MuiButtonBase-root-MuiCheckbox-root"
        data-testid="custom-checkbox-9"
      >
        <input
          checked=""
          class="PrivateSwitchBase-input css-1m9pwf3"
          data-indeterminate="false"
          id="custom-checkbox-9"
          name="Real Estate"
          type="checkbox"
        />
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="CheckBoxIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M19 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.11 0 2-.9 2-2V5c0-1.1-.89-2-2-2zm-9 14l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
          />
        </svg>
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </span>
      Real Estate
    </label>
    <label
      class="text-neutral-600 cursor-pointer"
    >
      <span
        class="MuiCheckbox-root MuiCheckbox-colorPrimary MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary PrivateSwitchBase-root Mui-checked css-12wnr2w-MuiButtonBase-root-MuiCheckbox-root"
        data-testid="custom-checkbox-10"
      >
        <input
          checked=""
          class="PrivateSwitchBase-input css-1m9pwf3"
          data-indeterminate="false"
          id="custom-checkbox-10"
          name="Energy"
          type="checkbox"
        />
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="CheckBoxIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M19 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.11 0 2-.9 2-2V5c0-1.1-.89-2-2-2zm-9 14l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
          />
        </svg>
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </span>
      Energy
    </label>
    <label
      class="text-neutral-600 cursor-pointer"
    >
      <span
        class="MuiCheckbox-root MuiCheckbox-colorPrimary MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary PrivateSwitchBase-root Mui-checked css-12wnr2w-MuiButtonBase-root-MuiCheckbox-root"
        data-testid="custom-checkbox-11"
      >
        <input
          checked=""
          class="PrivateSwitchBase-input css-1m9pwf3"
          data-indeterminate="false"
          id="custom-checkbox-11"
          name="Utilities"
          type="checkbox"
        />
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="CheckBoxIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M19 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.11 0 2-.9 2-2V5c0-1.1-.89-2-2-2zm-9 14l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
          />
        </svg>
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </span>
      Utilities
    </label>
  </div>
</div>
`;
